client/jetpack-cloud/sections/partner-portal/issue-license-v2/hooks/use-product-bundle-size.ts

```diff
@@ -1,21 +1,48 @@
-import { useCallback, useState } from 'react';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import useProductsQuery from 'calypso/state/partner-portal/licenses/hooks/use-products-query';
+import { APIProductFamilyProduct } from 'calypso/state/partner-portal/types';
 import { useURLQueryParams } from '../../hooks';
 
-const AVAILABLE_SIZES = [ 1, 5, 10, 20, 50, 75, 100 ]; //TO-DO: We will need to get this from the API
-
 const BUNDLE_SIZE_PARAM_KEY = 'bundle_size';
 
 // Parse the location hash to get the selected product bundle size
 // If the hash is not found, return the default size (1)
-const parseLocationHash = ( value: string ) => {
-	return AVAILABLE_SIZES.find( ( size ) => value === `${ size }` ) || 1;
+const parseLocationHash = ( supportedBundleSizes: number[], value: string ) => {
+	return supportedBundleSizes.find( ( size ) => value === `${ size }` ) || 1;
+};
+
+const getSupportedBundleSizes = ( products?: APIProductFamilyProduct[] ) => {
+	if ( products?.length ) {
+		return [
+			1,
+			...products.reduce( ( set, product ) => {
+				product.supported_bundles.forEach( ( { quantity } ) => {
+					set.add( quantity );
+				} );
+
+				return set;
+			}, new Set< number >() ),
+		];
+	}
+
+	return [ 1 ];
```

Comment:
Completely non-blocking, but I think this is a great spot to use `Array.flatMap`. What do you think of this sort of implementation?

```suggestion
	if ( ! products ) {
		return [ 1 ];
	}
	
	const supported = new Set( [
		1,
		...products.flatMap( ( p ) => p.supported_bundles ) )
	] );
	return [ ...supported ];
```

Reply:
Nice suggestion! 👍 It is more cleaner.